R-package/R/lgb.model.dt.tree.R

```diff
@@ -64,10 +63,16 @@ lgb.model.dt.tree <- function(model, num_iteration = NULL) {
   # Combine into single data.table fourth
   tree_dt <- data.table::rbindlist(tree_list, use.names = TRUE)
 
-  # Lookup sequence
-  tree_dt[, split_feature := Lookup(split_feature,
-                                    seq.int(from = 0, to = parsed_json_model$max_feature_idx),
-                                    parsed_json_model$feature_names)]
+  # Substitute feature index with the actual feature name
+
+  # Since the index comes from C++ (which is 0-indexed), be sure
+  # to add 1 (e.g. index 28 means the 29th feature in feature_names)
+  split_feature_indx <- tree_dt[, split_feature] + 1
+
+  # Get corresponding feature names. Positions in split_feature_indx
+  # which are NA will result in an NA feature name
+  feature_names <- parsed_json_model$feature_names[split_feature_indx]
+  tree_dt[, split_feaure := feature_names]
```

Comment:
WOW I think this is why the tests were failing for me on this build. typo: `split_feaure` 😆 

testing now